docs/Glossary.md

```diff
@@ -0,0 +1,141 @@
+
+## State
+
+_State_ (also called the _state tree_) is a broad term, but in the Redux API it usually refers to the single state value that is managed by the store and returned by [`getState()`](api/Store.md#getState) or the property syntax [`state`](api/Store.md#getState). It represents the entire state of a Redux application.
+
+State can be any type of object, however data classes are well suited due to their `copy()` method for creating the new state.  The contents and structure of the State are largly up to requirements of the app and the opinion of the implementer.  In general, avoiding deeply nested structure makes creating the new state easier.  Also this is the state of your App, not neccessarily the state of your UI. Still, you should do your best to keep the state serializable. Don't put anything inside it that you can't easily turn into JSON.
+
+## Action
+
+An _action_ is an object that represents an intention to change the state. Actions are the only way to get data into the store. Any data, whether from UI events, network callbacks, system events, or other sources such as WebSockets needs to eventually be dispatched as actions.
+
+ReduxKotlin differs a bit from JS Redux in that since we have a statically typed language, we can use the type as indication of the action to be performed. Typically this is done with a `when` statement:
+
+```kotlin
+	when (action) {
+	   is MyAction -> //handle MyAction
+```
+
+In JS redux, a `type` field is used to indicate the type of action performed.  This approach can be used with ReduxKotlin, however type safety and IDE benefits will be lost.
+
+The structure of an action object is really up to you.  Data classes are a good choice, since `toString()` is impemented and can be useful for logging. Data inside the action is used to update the state inside reducers.  For example, an API request result may be inside an Action.
+
+See also [async action](#async-action) below.
+
+## Reducer
+
+```kotlin
+typealias Reducer<State> = (state: State, action: Any) -> State
+```
+
+A _reducer_ (also called a _reducing function_) is a function that accepts an accumulation and a value and returns a new accumulation. They are used to reduce a collection of values down to a single value.
+
+Reducers are not unique to Redux—they are a fundamental concept in functional programming. Even most non-functional languages, like Kotlin, have a built-in API for reducing. In Kotlin, reduce is a[`function on collections and many types`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/reduce.html).
+
+In Redux, the accumulated value is the state object, and the values being accumulated are actions. Reducers calculate a new state given the previous state and an action. They must be _pure functions_—functions that return the exact same output for given inputs. They should also be free of side-effects. This is what enables exciting features like hot reloading and time travel.
```

Comment:
These exciting features should have links

Reply:
done.  also add comment about these are not quite ready for reduxkotlin